### ISSUE TITLE
Add --drop  option

### ISSUE THREAD

**Issue by Aaron2Ti** [watchexec/watchexec#123]
Hi sir,

`watchdog` has a pretty useful feature called `--drop`

https://github.com/gorakhargosh/watchdog/blob/845d76792a7eb8e81785f680e7af2d756d10f1ce/src/watchdog/watchmedo.py#L411
```
  -W, --drop            Ignore events that occur while command is still being
                        executed to avoid multiple simultaneous instances
                        (default: False)
```

Wondering if watchexec could support it in future please

Regards

**Comment by passcod** [comment]
That's the default behaviour. The inverse behaviour is toggled by `--restart`.

**Comment by Mange** [comment]
No, it's not. If I have a command that takes 10 seconds to run and it's started after my first change, it will immediately start a new job upon completion if I make another change during the runtime.

Try this "oneliner" in your shell to see the effect:

```sh
 watchexec --postpone \
  -- "echo '--- start ---'; date; sleep 3; date; echo '--- end ---'" &; \
  sleep 0.5; \
  echo ">>> First change"; echo foo > foo; \
  sleep 1; \
  echo ">>> Second change"; echo bar > foo; \
  sleep 5; \
  killall watchexec; wait
```

It'll start watchexec in the background, printing when a 3-second job starts and stops. A change is triggered after 0.5 seconds, which should show up as a single job run. However, in the middle of the job another change is made.

Without an option like this a second job trigger will start. The option above proposes that watchexec should only react to changes made while it "waits", not react when a change is happening while the command runs.

Here's the output from my shell:

```
[1] 2277
>>> First change
--- start ---
Fri 07 Jun 2019 04:02:22 PM CEST
>>> Second change
Fri 07 Jun 2019 04:02:25 PM CEST
--- end ---
--- start ---
Fri 07 Jun 2019 04:02:25 PM CEST
[1]  + terminated  watchexec --postpone -- 
```

**Comment by passcod** [comment]
Right, okay, but that's not really what the OP `--drop` is documenting. Watchexec never has simultaneous instances unless the child detaches, so something to avoid that is nonsensical.

I do agree your described behaviour would be good to have, though, and it's pretty much trivial to add, too. I'm just not sure about the name: `drop` doesn't feel particularly obvious?

**Comment by Mange** [comment]
Heh, I guess I focused too much on

> Ignore events that occur while command is still being executed […]

without reading the end of it. Maybe OP can clarify what they wanted specifically here.

I - on the other hand - wanted what I detailed above. If OP feels this is covered already due to the lack of multiple instances, I could open another issue for my wish.

---

> `drop` doesn't feel particularly obvious?

I agree. How about `--watch-on-idle`, `--watch-when-complete`, `--no-running-watch`, or `--pause-watch-when-running`? Shorter names could be `--idle-watch` or maybe even `--idle`?

Just putting some names out there.
